src/main/java/hudson/remoting/MultiClassLoaderSerializer.java

```diff
@@ -122,6 +124,7 @@ private ClassLoader readClassLoader() throws IOException, ClassNotFoundException
             channel.classFilter.check(name);
             try {
                 ClassLoader cl = readClassLoader();
+                // TODO: handle null classloader as a System one?
```

Comment:
8-year old change in `readClassLoader()`: https://github.com/jenkinsci/remoting/commit/6254e9a2e4e7fbd34bf2f75391768155f838163a
I cannot find any related issues